src/org/exist/launcher/Launcher.java

```diff
@@ -450,70 +381,15 @@ private void setServiceState() {
         }
     }
 
-    private void isRoot(Consumer<Boolean> consumer) {
-        final List<String> args = new ArrayList<>(2);
-        args.add("id");
-        args.add("-u");
-        run(args, (code, output) -> {
-            consumer.accept("0".equals(output.trim()));
-        });
-    }
-
-    private void showServicesConsole() {
-        final List<String> args = new ArrayList<>(2);
-        args.add("cmd");
-        args.add("/c");
-        args.add("services.msc");
-
-        run(args, null);
-    }
-
-    protected static void run(List<String> args, BiConsumer<Integer, String> consumer) {
-        final ProcessBuilder pb = new ProcessBuilder(args);
-        final Optional<Path> home = ConfigurationHelper.getExistHome();
-        if (home.isPresent()) {
-            pb.directory(home.get().toFile());
-        }
-        pb.redirectErrorStream(true);
-        try {
-            final Process process = pb.start();
-            if (consumer != null) {
-                final StringBuilder output = new StringBuilder();
-                try (final BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream(),
-                        "UTF-8"))) {
-                    String line;
-                    while ((line = reader.readLine()) != null) {
-                        output.append('\n').append(line);
-                    }
-                }
-                final int exitValue = process.waitFor();
-                consumer.accept(exitValue, output.toString());
-            }
-        } catch (IOException | InterruptedException e) {
-            JOptionPane.showMessageDialog(null, e.getMessage(), "Error Running Process", JOptionPane.ERROR_MESSAGE);
-        }
-    }
-
     protected void shutdown(final boolean restart) {
         utilityPanel.setStatus("Shutting down ...");
         SwingUtilities.invokeLater(() -> {
-            if (restart && runningAsService.isPresent()) {
-                final WrappedService service = runningAsService.get();
-                if (service.isRunning() || service.isStarting()) {
-                    if (service.stop()) {
-                        while (service.isRunning()) {
-                            try {
-                                wait(500);
-                            } catch (InterruptedException e) {
-                                break;
-                            }
-                        }
-                        if (!service.isRunning() && service.start()) {
-                            trayIcon.displayMessage(null, "Database restarted", TrayIcon.MessageType.INFO);
-                        } else {
-                            trayIcon.displayMessage(null, "Failed to restart. Please start service manually.", TrayIcon.MessageType.INFO);
-                        }
-                    }
+            if (serviceManager.isRunning()) {
```

Comment:
If I search for `serviceManager.` then I see several places where it looks to me like the internal state of `ServiceManager` is called from outside the AWT event dispatcher.